indexing/src/main/java/com/google/enterprise/cloudsearch/sdk/indexing/StructuredData.java

```diff
@@ -354,16 +359,21 @@ private NamedProperty getProperty(String propertyName, Collection<Object> values
       if (nonNullValues.isEmpty()) {
         return null;
       }
-      if (!isRepeated) {
-        return propertyBuilder.getNamedProperty(
-            propertyName, Collections.singletonList(valueConverter.convert(nonNullValues.get(0))));
-      } else {
-        return propertyBuilder.getNamedProperty(
-            propertyName,
-            nonNullValues
-                .stream()
-                .map(v -> valueConverter.convert(v))
-                .collect(Collectors.toList()));
+      try {
+        if (!isRepeated) {
+          return propertyBuilder.getNamedProperty(
+              propertyName, Collections.singletonList(valueConverter.convert(nonNullValues.get(0))));
+        } else {
+          return propertyBuilder.getNamedProperty(
+              propertyName,
+              nonNullValues
+                  .stream()
+                  .map(v -> valueConverter.convert(v))
+                  .collect(Collectors.toList()));
+        }
+      } catch (IllegalArgumentException e) {
+        if(ignoreConversionErrors) return null;
```

Comment:
I would definitely add a log here. Having to check for every single result if a property is there without being warned this happen will be a pain.